app/models/iteration.rb

```diff
@@ -7,9 +7,15 @@ class Iteration < ApplicationRecord
 
   has_one :exercise, through: :solution
   has_one :track, through: :exercise
+  has_one :user, through: :solution
   has_one :test_run, through: :iteration
 
   scope :not_deleted, -> { where(deleted_at: nil) }
+  scope :latest, lambda {
```

Comment:
I made the latest iterations logic into a scope, which makes it easier to unit test and use.

Reply:
Really nice.